File: mdd-remote/src/main/java/io/mateu/remote/domain/queries/getJourney/GetJourneyQueryHandler.java
package io.mateu.remote.domain.queries.getJourney;

import io.mateu.remote.domain.store.JourneyStoreService;
import io.mateu.remote.dtos.Journey;
import lombok.extern.slf4j.Slf4j;
import org.springframework.stereotype.Service;

@Service
@Slf4j
public class GetJourneyQueryHandler {

    public Journey run(GetJourneyQuery query) throws Exception {

        String journeyId = query.getJourneyId();

        return JourneyStoreService.get().getJourney(journeyId);

    }
}

File: frontend/src/mateu/journey/domain/queries/getJourney/GetJourneyQueryHandler.ts
import {GetJourneyQuery} from "./GetJourneyQuery";
import Journey from "../../../../shared/apiClients/dtos/Journey";
import {mateuApiClient} from "../../../../shared/apiClients/MateuApiClient";

export class GetJourneyQueryHandler {

    public async handle(_query: GetJourneyQuery): Promise<Journey> {
        return await mateuApiClient
            .fetchJourney(_query.journeyTypeId, _query.journeyId)
    }

}

export const getJourneyQueryHandler = new GetJourneyQueryHandler()